### PR TITLE
Adding IsRequired property to all parameters

### PR DESCRIPTION
**Pull request by teresaqhoang**
### Motivation and Context

Extending parameter objects to include an `IsRequired` field for better planner resiliency.

### Description

Also, includes some logic changes to Copilot PlanViewer to show INPUT value.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [x] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs [hunk 1]
```diff
@@ -16,8 +16,10 @@ internal static string ToManualString(this FunctionView function)
     {
         var inputs = string.Join("\n", function.Parameters.Select(parameter =>
         {
-            var defaultValueString = string.IsNullOrEmpty(parameter.DefaultValue) ? string.Empty : $" (default value: {parameter.DefaultValue})";
-            return $"  - {parameter.Name}: {parameter.Description}{defaultValueString}";
+            var requiredString = parameter.IsRequired ? "required" : string.Empty;
+            var defaultValueString = string.IsNullOrEmpty(parameter.DefaultValue) ? string.Empty : $"default value: {parameter.DefaultValue}";
+            var detailsString = $" ({AddComma(requiredString, defaultValueString)})";
```
Comment (lemillermicrosoft) [inline dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs:21]: Could `string.Join` accomplish the same thing?

File: dotnet/src/SemanticKernel.Abstractions/SkillDefinition/SKFunctionContextParameterAttribute.cs [hunk 2]
```diff
@@ -41,6 +41,11 @@ public string Name
...
     public string DefaultValue { get; set; } = string.Empty;
 
+    /// <summary>
+    /// Whether the parameter is required.
+    /// </summary>
+    public bool IsRequired { get; set; } = true;
+
     /// <summary>
     /// Creates a parameter view, using information from an instance of this class.
     /// </summary>
```
Comment (lemillermicrosoft) [inline dotnet/src/SemanticKernel.Abstractions/SkillDefinition/SKFunctionContextParameterAttribute.cs:51]: #1195 is proposing some changes in the area. Might use this to help speed up things over there.

**Review by adrianwyatt — APPROVED** [review]
LGTM - but I'm curious about using string.join as well.

File: dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs (hunk 1 above)
Comment (teresaqhoang) [reply dotnet/src/Extensions/Planning.SequentialPlanner/FunctionViewExtensions.cs:21]: Nah string.Join doesn't consider if an element is non-empty without an explicit check

image

fyi @adrianwyatt

**Review by amsacha — APPROVED** [review]

**Comment by teresaqhoang** [comment]
Closing in favor of Stephen's PR: https://github.com/microsoft/semantic-kernel/pull/1195 

Will an issue to re-evaluate how we specify required / optional parameters to the Planner. It seems like Stephen's PR will handle native functions well, where all parameters are required but have a DefaultValue of null when optional. But some native functions specify (optional) in the parameter description, whereas we're specifying (required) in the description of rest api parameters when necessary. The AI could get confused for parameters with descriptions that don't contain either postfix, so we should pick one flag or the other for the planner's functionView.

**Pull request closed without merging**